### PR TITLE
fix json unmarshal error when list plguins

### PR DESCRIPTION
**Pull request by cmssczy**
### Type of change:

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### What this PR does / why we need it:

bug fix https://github.com/apache/apisix-ingress-controller/issues/883
the response body is JSON with unknown field names but not a string slice.
### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by tao12345666333** [comment]
Thanks! It's on my list

File: pkg/apisix/cluster.go [hunk 1]
```diff
@@ -750,10 +750,16 @@ func (c *cluster) getList(ctx context.Context, url, resource string) ([]string,
...
 	}
 
-	var listResponse []string
+	var listResponse map[string]interface{}
 	dec := json.NewDecoder(resp.Body)
 	if err := dec.Decode(&listResponse); err != nil {
 		return nil, err
 	}
-	return listResponse, nil
+	var res []string = make([]string, len(listResponse))
```
Comment (tokers) [inline pkg/apisix/cluster.go:758]:
```suggestion
	res []string := make([]string, 0, len(listResponse))
```

After that, just using `append`, we don't need the indice `i`.

**Review by tokers — COMMENTED** [review]
@cmssczy Thanks for your PR, could you give us an example about the response? Fair to say, I didn't realize the difference.

**Comment by cmssczy** [comment]
> could you give us an example about the response? Fair to say, I didn't realize the difference.

@tokers 
The response body json format is like:
```json
{
	"serverless-pre-function": {
		"version": 0.1,
		"priority": 10000
	},
	"echo": {
		"version": 0.1,
		"priority": 412
	}
}
```
It is a json struct with unknown field names but not a string slice. So I think we should use a `map` instead of `slice`.

File: pkg/apisix/cluster.go (hunk 1 above)
Comment (cmssczy) [reply pkg/apisix/cluster.go]: Done

**Review by tao12345666333 — APPROVED** [review]
LGTM

**Comment by lingsamuel** [comment]
We need a mechanism to ensure that the APISIX interface we are using is not outdated.

**Comment by tao12345666333** [comment]
> We need a mechanism to ensure that the APISIX interface we are using is not outdated.

yep. we may need to convert directly from the upstream schema.

now let's move forward.

**Pull request merged**